helpers/errors.go

```diff
@@ -0,0 +1,16 @@
+package helpers
+
+type TPError struct {
+	str  string
+	code int
```

Comment:
Lowercase-starting names mean the fields are invisible outside the package, and we'll probably want to get the error code in the controllers. Change `code` to `Code`.